core/src/test/java/org/infinispan/test/FeaturesListener.java

```diff
@@ -0,0 +1,84 @@
+package org.infinispan.test;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.util.List;
+
+import org.infinispan.commons.util.Features;
+import org.infinispan.test.fwk.NamedTestMethod;
+import org.testng.IMethodInstance;
+import org.testng.IMethodInterceptor;
+import org.testng.ITestClass;
+import org.testng.ITestContext;
+import org.testng.ITestNGMethod;
+import org.testng.internal.BaseTestMethod;
+
+public class FeaturesListener implements IMethodInterceptor {
+
+   @Override
+   public List<IMethodInstance> intercept(List<IMethodInstance> methods, ITestContext context) {
+      Features features = new Features(this.getClass().getClassLoader());
```

Comment:
Maybe we should use the class loader of the test instance which you are retrieving in the line below.